esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.h

```diff
@@ -0,0 +1,2145 @@
+        switch (frame.state)
+        {
+        case DataFrame::New:
+            fprintf(stdout, "New data frame\n");
+            return;
+        case DataFrame::Pulled:
+            fprintf(stdout, "Pulled");
+            break;
+        case DataFrame::TypeReported:
+            fprintf(stdout, "TypeReported");
+            break;
+        case DataFrame::DataReported:
+            fprintf(stdout, "DataReported");
+            break;
+        case DataFrame::FrameIgnored:
+            fprintf(stdout, "FrameIgnored");
+            break;
+        default:
+            fprintf(stdout, "unknown (%d)", frame.state);
+        }
+        fprintf(stdout, " data frame: ");
+        switch (frame.eventType)
+        {
+        case START_TAG:
+            fprintf(stdout, "<%s>", frame.stag.getQName());
+            break;
+        case CONTENT:
+            fprintf(stdout, "<![CDATA[%s]]>", frame.content);
+            break;
+        case END_TAG:
+            fprintf(stdout, "</%s>", frame.etag.getQName());
+            break;
+        case END_DOCUMENT:
+            fprintf(stdout, "EOF");
+            break;
+        default:
+            fprintf(stdout, "unknown (%d)", frame.eventType);
+            break;
+        }
+        fprintf(stdout, "\n");
+    }
+
+};
+
+/**
+ * IFragmentedXmlAssistant implementation supporting the registration of elements that may
+ * represent either external or embedded fragments.
+ * 
+ * For embedded fragments, the assistant recognizes which elements may contain embedded fragment
+ * data, which of those do contain embedded fragment data, and creates the fragment representing
+ * the embedded fragment data.
+ *
+ * For external fragments, the assistat recognizes which elements may contain external fragment
+ * references. For elements that may contain references, it relies upon a separate detector to
+ * determine if they do, in fact, contain an external reference. For elements that do contain
+ * external references, it relies on a separate loader to resolve the detected key(s) into a
+ * fragment with the requested data.
+ */
+class CFragmentedXmlAssistant : public CInterfaceOf<IFragmentedXmlAssistant>
+{
+public:
+    enum DetectedType
+    {
+        Unknown,
+        Resource,
+        Path,
+    };
+
+    /**
+     * Abstraction representing the "identity" of an external fragment. The identity of a fragment
+     * is the information located by a detector that can be used by a loader to get the fragment
+     * data. It is assumed that a fragment may be uniquely identified by at least one key and at
+     * most two keys. The first, or primary, key may be a file path or resource bundle identifier.
+     * The second, if used, may be an XPath referring to a nodeset within a broader XML document.
+     */
+    interface IExternalDetection : extends IInterface
+    {
+        virtual int         queryDetectedType() const = 0;
+        virtual const char* queryPrimaryKey() const = 0;
```

Comment:
Not a fan of calling something that might be a filepath a "key" even if it may be a look up when loading from a resource pool.  Id rather either be explicit with these function names... queryFilePath() or only have one queryLocation() and have the meaning depend on the type.